provider/keycloak_role_helpers.go

```diff
@@ -0,0 +1,47 @@
+package provider
+
+import "github.com/mrparkers/terraform-provider-keycloak/keycloak"
+
+func getMapOfRealmAndClientRoles(keycloakClient *keycloak.KeycloakClient, realmId string, roleIds []string) (map[string][]*keycloak.Role, error) {
+	roles := make(map[string][]*keycloak.Role)
+
+	for _, roleId := range roleIds {
+		role, err := keycloakClient.GetRole(realmId, roleId)
+		if err != nil {
+			return nil, err
+		}
+
+		if role.ClientRole {
+			roles[role.ClientId] = append(roles[role.ClientId], role)
+		} else {
+			roles["realm"] = append(roles["realm"], role)
```

Comment:
I see you use "realm" as a keyword in the map of roles. But what if there is a client called "realm"?
Is there a reason for not using the UserRoleMapping struct instead of this map of roles?

Reply:
Correct, I moved this function to reuse it. And I reused this "pattern" with 'realm' as a special-keywork for clients without challenging it.
But looking at the code, using a dedicated struct might simplify the handling/code, and would avoid the edge-case with clients named "realm".

I will look into this and see if I can change this.

Reply:
Yeah feel free to blame me for this one.  Using a dedicated struct is probably better, but I won't force you to fix this since it was my decision to do this in the first place.